### PR TITLE
Add new spacing scale

### PR DESCRIPTION
**Pull request by kstinson14**
Over the past few weeks, some of us from the calcite + field apps team met to discuss the new spacing scale, and decided to try out using the [Tailwind](https://tailwindcss.com/docs/padding) scale.  I've implemented that here so that we can start to propagate this out to other repos (like calcite-components).

### CODE REVIEW

File: dist/_index.scss [hunk 1]
```diff
@@ -1,7 +1,6 @@
 @import "./_variables.scss";
 @import "./_animation.scss";
 @import "./_responsive.scss";
-@import "./_spacing.scss";
```
Comment (kstinson14) [inline dist/_index.scss:4]: I decided to remove the existing spacing mixins, as I'm under the impression they won't be needed with the new scale.

File: dist/index.js [hunk 2]
```diff
@@ -19,9 +18,29 @@ export const avenirTracking = "0";
...
-export const shadow2 = "0 0 16px 0 rgba(0,0,0,0.15)"; 
-export const shadow2Hover = "0 0 20px 0 rgba(0,0,0,0.2)"; 
-export const shadow2Press = "0 0 20px 0 rgba(0,0,0,0.3)"; 
+export const shadow1 = "0 0 8px 0 rgba(0,0,0,0.08)"; 
+export const shadow1Hover = "0 0 16px 0 rgba(0,0,0,0.08)"; 
+export const shadow1Press = "0 0 8px 0 rgba(0,0,0,0.16)"; 
+export const shadow2 = "0 0 16px 0 rgba(0,0,0,0.16)"; 
+export const shadow2Hover = "0 0 32px 0 rgba(0,0,0,0.16)"; 
+export const shadow2Press = "0 0 16px 0 rgba(0,0,0,0.32)"; 
+export const borderRadius = "3px";
```
Comment (kstinson14) [inline dist/index.js:27]: After running `npm start` or maybe `npm docs`, I did get a few changes that I didn't expect (such as the ones above to the shadow variables and the borderRadius addition.  I'm not sure if this is desired or not, it _does_ seem like the shadow variables are set to these new values elsewhere (`variables.scss`), so I left them in.

**Comment by paulcpederson** [comment]
Nice, thanks @kstinson14 . We should note this change as breaking in the `CHANGELOG`, maybe under an "Unreleased" section?

**Comment by macandcheese** [comment]
Rad! Thanks. Do you all have any preference for nomenclature should utility classes get generated from these?

**Comment by kstinson14** [comment]
For utility classes, we do really like the default [Tailwind](https://tailwindcss.com/docs/padding/) format.  For spacing, this follows the format of:
`[letter symbolizing property][letter symbolizing direction (optional)]-[spacing value]`
Examples:

`p-4` = padding of spacing value `4` on all sides of element
`pl-4` = padding of spacing value `4` on left side of element
`pr-4` = padding of spacing value `4` on left side of element
`px-4` = padding of spacing value `4` on left / right sides of element
`py-4` = padding of spacing value `4` on top / bottom sides of element

Margin follows the same pattern.  Width and height follow a similar pattern with classes such as `w-16` or `h-2` denoting a "width of spacing value `16`" or a "height of spacing value `2`".

It's very succinct but still easy to parse, so we enjoy it quite a bit.

**Comment by kstinson14** [comment]
@paulcpederson @julio8a @macandcheese good to merge this one?

**Comment by paulcpederson** [comment]
Good by me, I think @macandcheese has the yellow dot on this one

**Pull request merged**